pkg/subscriber/github/git_subscriber_item.go

```diff
@@ -323,29 +328,30 @@ func (ghsi *SubscriberItem) subscribeResourceFile(hostkey types.NamespacedName,
 	pkgMap[dpltosync.GetName()] = true
 
 	klog.V(4).Info("Ready to register template:", hostkey, dpltosync, syncsource)
+	err = ghsi.synchronizer.AddTemplates(syncsource, hostkey, []kubesynchronizer.DplUnit{{Dpl: dpltosync, Gvk: *validgvk}})
 
-	err = ghsi.synchronizer.RegisterTemplate(hostkey, dpltosync, syncsource)
 	if err != nil {
 		err = utils.SetInClusterPackageStatus(&(ghsi.Subscription.Status), dpltosync.GetName(), err, nil)
 
+		ghsi.successful = false
+
 		if err != nil {
 			klog.V(4).Info("error in setting in cluster package status :", err)
 		}
 
-		pkgMap[dpltosync.GetName()] = true
-
 		return err
 	}
 
 	dplkey := types.NamespacedName{
 		Name:      dpltosync.Name,
 		Namespace: dpltosync.Namespace,
 	}
-	kvalid.AddValidResource(*validgvk, hostkey, dplkey)
 
 	pkgMap[dplkey.Name] = true
+	//this is for the adaption of the new synchronizer API
+	klog.V(10).Infof("ingnor this kvalid %v, only exist for adopting synchronizer API", kvalid)
 
-	return nil
+	return err
```

Comment:
Double check please. Always return err if it gets this far?

Reply:
you are right. I should've use `return nil`, which is more friendly. I don't remember why I made this change...

However, using `return err` is equal to `return nil` over here. I mean, if the code gets this far, the only way is the `err == nil`

Reply:
Ok you should change it back to return nil in case err is set in the future and code readability.

Reply:
sure